compiler/src/dmd/tocsym.d

```diff
@@ -339,6 +342,39 @@ Symbol *toSymbol(Dsymbol s)
             const(char)* id = mangleExact(fd);
 
             //printf("FuncDeclaration.toSymbol(%s %s)\n", fd.kind(), fd.toChars());
+
+            /* MS-LINK cannot handle multiple COMDATs with the same name.
+             * This can happen with two .c files that define the same function.
+             * They get compiled into one .obj file with the `oneobj` setting.
+             * https://issues.dlang.org/show_bug.cgi?id=24129
+             */
+            if (driverParams.oneobj)
+            {
+                auto mod = fd.getModule();
+                if (mod &&
+                    mod.isRoot() &&               // included on command line
+                    mod.filetype == FileType.c && // a C file
+                    fd.fbody &&                   // a function definition
+                    fd._linkage == LINK.c &&
+                    !fd.skipCodegen)              // code gen is desired
+                {
+                    __gshared DsymbolTable Csymtab;  // sorry about another global variable
```

Comment:
Does it have to be declared inline? The rest of the compiler has a more (...) global pattern for the global variables.

Reply:
@WalterBright can we make `Csymtab` a member variable of the `ToSymbol` visitor class? If not, why? Perhaps we can move it up the call stack and pass it as a parameter to the visitor constructor, in case there are multiple instances?

Reply:
I prefer to prove that this fix works. Refactoring it then can (and should) be an independent task.